modules/nf-core/vcftools/main.nf

```diff
@@ -126,7 +126,71 @@ process VCFTOOLS {
     """
 
     stub:
+    def prefix = task.ext.prefix ?: "${meta.id}"
     """
+    touch ${prefix}.vcf
```

Comment:
Perhaps a silly question, but should all those files always be written regardless of the "arguments" for vcftools?

Reply:
I was also wondering about that. We would need to collect the command and then produce the files depending on that. That is definetly a more complicated stub than what I was aiming for.

Reply:
Yes, I guess it would be a lot more complicated, and I'm inclined to approve as-is.

I just happened to take a look [here](https://github.com/nf-core/modules/blob/4bd9481c5718ca7bfd31459b8242c349171dac87/modules/nf-core/bwa/mem/main.nf#L59). There you see `samtools_command` being defined in the stub, but AFAICT, it is totally irrelevant to the "code" be executed in the stub.

Reply:
As I understand it, most of the times every `def` statement can be removed when defining the stub. Except for 
`    def prefix = task.ext.prefix ?: "${meta.id}"`

seems like they do not differe there as well and just produce all outputs